controllers/cron_rollout_helpers.go

```diff
@@ -120,13 +126,29 @@ func createBaseDisruption(owner metav1.Object, disruptionSpec *chaosv1beta1.Disr
 }
 
 // setDisruptionAnnotations updates the annotations of a given Disruption object with those of its owner.
-// Additionally, it sets a scheduled time annotation using the provided scheduledTime.
-func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time) {
-	for k, v := range owner.GetAnnotations() {
+// It sets a scheduled time annotation using the provided scheduledTime.
+// It parses the UserInfo annotation if it exists and sets user-related annotations.
+func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time, log *zap.SugaredLogger) {
+	// Copy owner annotations to Disruption
```

Comment:
Why are we copying all the owner annotations to the child?

Reply:
The copying of owner's annotation is something that was already present. I believe it is done to maintain metadata consistency across related resources. This ensures that any important metadata, such as scheduling, monitoring, or custom annotations, is preserved when creating the Disruption from the DisruptionCron or DisruptionRollout.